core/commands/dns.go

```diff
@@ -63,7 +63,7 @@ The resolver will give:
 			depth = namesys.DefaultDepthLimit
 		}
 		output, err := resolver.ResolveN(req.Context().Context, name, depth)
-		if err != nil {
+		if err != nil && (err != namesys.ErrResolveRecursion || depth > 1) {
```

Comment:
On Wed, Jun 10, 2015 at 01:26:47AM -0700, Juan Batiz-Benet wrote:
> > -		if err != nil {
> > +		if err != nil && (err != namesys.ErrResolveRecursion || depth > 1) {
> 
> Is the goal to do something like `io.EOF` which is returned even
> when there is valid data returned?

Yeah, I guess that's a similar idea.  In any case, we're saying
“here's how far we got: <path>”, and then either “and it was an
unequivocal success (nil error)” or “we have some concerns about our
success: <err>” where the concerns could be:

* A resolution step failed.  You should probably just give up.
* All the resolution steps succeeded, but we hit our allotted
  recursion limit before finding an /ipfs/… path.  You can give up if
  you like, but if this is really important to you, you can try
  another round of resolution or increase your depth limit in the
  future, and you might eventually resolve this path to /ipfs/….